addons/hr_attendance/models/hr_attendance.py

```diff
@@ -92,71 +92,47 @@ def _compute_color(self):
 
     @api.depends('worked_hours')
     def _compute_overtime_hours(self):
-        att_progress_values = dict()
-        negative_overtime_attendances = defaultdict(lambda: False)
         if self.employee_id:
             self.env['hr.attendance'].flush_model(['worked_hours'])
             self.env['hr.attendance.overtime'].flush_model(['duration'])
```

Comment:
When testing I found that it was needed but I'm not sure I fully understand why?
When we call it from `_update_overtime` the transaction isn't over or something similar was my understanding but I'm not sure why it doesn't work with the cache?

Reply:
Normally, an explicit flush with no queries indicate a problem with the dependencies of the computes.
I see that the method you've mentionned `_update_overtime` does manual triggering of computes, hence why maybe you found it needed?
IMHO It would be interesting to get to the end of it to understand the *why*. `flush_query` in the ORM is supposed to flush the necessary fields extracted from the search domain when doing the query.

Reply:
After testing a bit apparently the flush **wasn't** needed regardless of `search` or `search_fetch` so I'm guessing it was during an intermediate stage or i just got it wrong.